### PR TITLE
Jordan Sewell

### PR DESCRIPTION
**Pull request by jrsewell400**
Able to get 45% through the extension :) :) :)

### CODE REVIEW

File: app/models/professor.rb [hunk 1]
```diff
@@ -1,3 +1,10 @@
 class Professor <ApplicationRecord
+  validates_presence_of :name, :age, :specialty
 
+  has_many :professor_students
+  has_many :students, through: :professor_students
+
+  def average_age
+    students.average(:age)
```
Comment (mikedao) [inline app/models/professor.rb]: great

File: app/models/student.rb [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
 
+  has_many :professor_students
+  has_many :professors, through: :professor_students 
+
+  def count 
+    professors.count
+  end
+
+  def self.sort
+    order(name: :asc)
```
Comment (mikedao) [inline app/models/student.rb]: i think id call it something more descriptive than sort

File: spec/models/professor_spec.rb [hunk 3]
```diff
@@ -11,4 +11,48 @@
...
+      ProfessorStudent.create!(student: harry, professor: snape)
+      ProfessorStudent.create!(student: harry, professor: hagarid)
+      ProfessorStudent.create!(student: harry, professor: lupin)
+      ProfessorStudent.create!(student: malfoy, professor: hagarid)
+      ProfessorStudent.create!(student: longbottom, professor: lupin)
+      ProfessorStudent.create!(student: longbottom, professor: snape)
+
+      expect(snape.average_age).to eq(20)
+      expect(hagarid.average_age).to eq(15)
+      expect(lupin.average_age).to eq(20)
```
Comment (mikedao) [inline spec/models/professor_spec.rb]: great

File: spec/models/professor_spec.rb [hunk 4]
```diff
@@ -11,4 +11,48 @@
...
+    #   longbottom = Student.create!(name: "Neville Longbottom" , age: 30 , house: "Gryffindor" )
+
+    #   ProfessorStudent.create!(student: harry, professor: snape)
+    #   ProfessorStudent.create!(student: harry, professor: hagarid)
+    #   ProfessorStudent.create!(student: harry, professor: lupin)
+    #   ProfessorStudent.create!(student: malfoy, professor: hagarid)
+    #   ProfessorStudent.create!(student: longbottom, professor: lupin)
+    #   ProfessorStudent.create!(student: longbottom, professor: snape)
+
+    #   Student.order(name: :asc).all.to_a == [harry, longbottom, malfoy]
```
Comment (mikedao) [inline spec/models/professor_spec.rb]: whats this?

Comment (jrsewell400) [reply spec/models/professor_spec.rb]: So I copied over my test for the Student class method "sort" into Professor model test.  I assumed we'd need a class method/test for both of these resources.  Was this the right way to handle this, or is this violating DRY principles?  Are we able to do the sorting for both index page using one model method?

Also, I was able to finish the rest of the extension!  Woop!
https://github.com/jrsewell400/hogwarts_again

**Pull request closed without merging**